### PR TITLE
[JUJU-576] Update upstream deps, including using txn/v2 and worker/v3

### PR DESCRIPTION
**Pull request by wallyworld**
Update a bunch of juju upstream repo dependencies.
Start using juju/worker/v3 and juju/txn/v2. 

Changes are import churn.

## QA steps

juju bootstrap
juju deploy ubuntu

### CODE REVIEW

File: state/application.go [hunk 1]
```diff
@@ -12,19 +12,20 @@ import (
...
 	"github.com/juju/names/v4"
 	"github.com/juju/schema"
-	jujutxn "github.com/juju/txn"
+	jujutxn "github.com/juju/txn/v2"
 	"github.com/juju/utils/v3"
 	"github.com/juju/version/v2"
 	"gopkg.in/juju/environschema.v1"
 
+	csparams "github.com/juju/charmrepo/v6/csclient/params"
+
```
Comment (hpidcock) [inline state/application.go:28]: remove new line

File: state/binarystorage/binarystorage_test.go [hunk 2]
```diff
@@ -10,15 +10,16 @@ import (
...
 	jc "github.com/juju/testing/checkers"
-	jujutxn "github.com/juju/txn"
-	txntesting "github.com/juju/txn/testing"
+	jujutxn "github.com/juju/txn/v2"
+	txntesting "github.com/juju/txn/v2/testing"
 	"github.com/juju/version/v2"
 	gc "gopkg.in/check.v1"
 
+	"github.com/juju/blobstore/v2"
+
```
Comment (hpidcock) [inline state/binarystorage/binarystorage_test.go:22]: remove NL

File: state/export_test.go [hunk 3]
```diff
@@ -22,13 +21,15 @@ import (
...
 	"github.com/juju/names/v4"
 	jc "github.com/juju/testing/checkers"
-	jujutxn "github.com/juju/txn"
-	txntesting "github.com/juju/txn/testing"
+	jujutxn "github.com/juju/txn/v2"
+	txntesting "github.com/juju/txn/v2/testing"
 	jutils "github.com/juju/utils/v3"
 	"github.com/juju/worker/v3"
 	"github.com/kr/pretty"
 	gc "gopkg.in/check.v1"
```
Comment (hpidcock) [inline state/export_test.go:30]: remove NL

**Review by hpidcock — APPROVED** [review]

**Comment by wallyworld** [comment]
$$merge$$

**Comment by wallyworld** [comment]
$$merge$$

**Pull request merged**